dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/DefaultTrackerOwnershipManager.java

```diff
@@ -254,15 +253,20 @@ public boolean hasAccess( User user, TrackedEntityInstance entityInstance, Progr
             return true;
         }
 
-        OrganisationUnit ou = getOwner( entityInstance.getId(), program, entityInstance::getOrganisationUnit );
+        OrganisationUnit ownerOrgUnit = getOwner( entityInstance.getId(), program );
+        boolean hasOwner = ownerOrgUnit != null;
 
-        if ( program.isOpen() || program.isAudited() )
+        if ( program.isOpen() || program.isAudited() || !hasOwner )
```

Comment:
Maye it will be more readable extracting the if (!hasOwner logic) so having it as a guard clause and then another if(!program.isOpen() && !program.isAudited())